src/activity_item_handling.cpp

```diff
@@ -528,6 +531,9 @@ void activity_handlers::washing_finish( player_activity *act, Character *you )
     comps1.emplace_back( itype_soap, required.cleanser );
     comps1.emplace_back( itype_detergent, required.cleanser );
     you->consume_items( comps1 );
+    comps1.clear();
+    comps1.emplace_back( itype_liquid_soap, required.cleanser );
+    you->consume_items( comps1, 1, is_liquid_crafting_component );
```

Comment:
```suggestion
    comps1.emplace_back( itype_liquid_soap, required.cleanser );
    you->consume_items( comps1 );
```

Reply:
Calling `consume_item()` twice is trying to use soap or detergent on the first call (up to the required charges) AND trying to use liquid soap afterwards. The first call is failing because it can't find any detergent/soap. If it did, it would consume both the soap/detergent and the liquid soap.

By adding the liquid soap as an extra component it will try to use one of the 3 kinds of item in a single batch. 1 is the default batch size and `is_liquid_crafting_component` seems to be a filter that only returns true for liquids or items that aren't counted by charges. Since the default filter always returns true, using the default values should try to use any of the components that you added to the comps1 list.